src/main/java/com/aws/greengrass/lifecyclemanager/Kernel.java

```diff
@@ -163,6 +163,23 @@ public Kernel() {
         context.put(SERVICE_TYPE_TO_CLASS_MAP_KEY, typeToClassMap);
     }
 
+    /**
+     * Construct the Kernel to be used for connectivity validation.
+     * @param currentConfig a Map of the current configs being used by the device
+     * @param deploymentConfig a Map of the configs being merged in from deployment
+     */
+    public Kernel(Map<String, Object> currentConfig, Map<String, Object> deploymentConfig) {
+        context = new Context();
+        config = new Configuration(context);
+        config.mergeMap(System.currentTimeMillis(), currentConfig);
+        config.mergeMap(System.currentTimeMillis(), deploymentConfig);
```

Comment:
is the timestamp here correct? I think you need deployment creation timestamp vs existing timestamp?

Reply:
Correct me if I'm wrong, my understanding is that since this is a new config it has a timestamp of 0, so I can simply use the current time to merge in existing configs followed by the new configs to create what we want to validate against

Reply:
new configs do not always override existing configs in actual deployment config merge right? It compares the deployment creation timestamp against the existing config timestamp. 

https://github.com/aws-greengrass/aws-greengrass-nucleus/blob/82039a1d8a1fe81d215e2578614d5d621c7911f2/src/main/java/com/aws/greengrass/deployment/activator/DeploymentActivator.java#L109

Reply:
I will take a look at using the correct timestamps